src/service.cpp

```diff
@@ -137,8 +137,17 @@ class DisplayIndicator::Impl
 
             if (pSchema != NULL)
             {
+                gboolean bColorScheme = g_settings_schema_has_key (pSchema, "color-scheme");
                 g_settings_schema_unref (pSchema);
-                pColorSchemeSettings = g_settings_new (sSchema);
+
+                if (bColorScheme)
+                {
+                    pColorSchemeSettings = g_settings_new (sSchema);
+                }
+                else
+                {
+                    g_warning ("org.gnome.desktop.interface::color-scheme not found. Native theme profile changes will not be triggered.");
```

Comment:
Personally, I think we should check presence of the gsettings at the very first and, based on this, decide whether to offer the Theme Profile submenu or not.

Reply:
But we do. This only skips invoking the native Gtk method. Switching profiles with two themes still works.